cpp/arcticdb/python/python_handlers.cpp

```diff
@@ -6,25 +6,46 @@
  */
 #include <python/python_handlers.hpp>
 #include <python/gil_lock.hpp>
+#include <arcticdb/codec/codec.hpp>
 
 namespace arcticdb {
-    void EmptyHandler::handle_type(
-        const uint8_t*&,
-        uint8_t* dest,
-        const VariantField&,
-        const entity::TypeDescriptor&,
-        size_t dest_bytes,
-        std::shared_ptr<BufferHolder>
-    ) {
-        ARCTICDB_SAMPLE(HandleEmpty, 0);
-        util::check(dest != nullptr, "Got null destination pointer");
-        const size_t num_rows = dest_bytes / get_type_size(DataType::EMPTYVAL);
-        static_assert(get_type_size(DataType::EMPTYVAL) == sizeof(PyObject*));
-        const PyObject** ptr_dest = reinterpret_cast<const PyObject**>(dest);
-        py::none none = {};
-        for(auto row = 0u; row < num_rows; ++row) {
-            none.inc_ref();
-            *ptr_dest++ = none.ptr();
-        }
+
+void EmptyHandler::handle_type(
+    const uint8_t *&input,
+    uint8_t *dest,
+    const VariantField &variant_field,
+    const entity::TypeDescriptor &,
+    size_t dest_bytes,
+    std::shared_ptr<BufferHolder>
+) {
+    ARCTICDB_SAMPLE(HandleEmpty, 0)
+    util::check(dest != nullptr, "Got null destination pointer");
+    const size_t num_rows = dest_bytes / get_type_size(DataType::EMPTYVAL);
+    static_assert(get_type_size(DataType::EMPTYVAL) == sizeof(PyObject *));
+    auto ptr_dest = reinterpret_cast<const PyObject **>(dest);
+    py::none none = {};
+    for (auto row = 0u; row < num_rows; ++row) {
+        none.inc_ref();
+        *ptr_dest++ = none.ptr();
     }
+
+    util::variant_match(variant_field, [&input](const auto &field) {
+        using EncodedFieldType = std::decay_t<decltype(*field)>;
+        if constexpr (std::is_same_v<EncodedFieldType, arcticdb::EncodedField>)
```

Comment:
Is this going to be correct always? Maybe we should propagate the encoding version from the segment and check on it.

Reply:
Good question. The encoding version is really a write-time thing, it controls which encoding you would prefer to write in. Clients with a write preference for one or the other should still be able to read either version, and the generally-used proxy for that is exactly this type distinction.